cloud/scope/machinepool.go

```diff
@@ -405,10 +405,36 @@ func (m *MachinePoolScope) InstanceGroupTemplateBuilder(bootstrapData string) *c
 	instanceTemplate.Properties.ServiceAccounts = append(instanceTemplate.Properties.ServiceAccounts, m.InstanceServiceAccountsSpec())
 	instanceTemplate.Properties.NetworkInterfaces = append(instanceTemplate.Properties.NetworkInterfaces, m.InstanceNetworkInterfaceSpec())
 	instanceTemplate.Properties.Metadata.Items = append(instanceTemplate.Properties.Metadata.Items, m.InstanceAdditionalMetadataSpec()...)
+	instanceTemplate.Properties.ShieldedInstanceConfig = m.GetShieldedInstanceConfigSpec()
 
 	return instanceTemplate
 }
 
+// GetShieldedInstanceConfigSpec returns the shielded config spec for the instance
+// As of now will only build the configuration for using
+// - Integrity Monitoring - enabled by default
+// - Secure Boot - disabled by default
+// - vTPM - enabled by default
+func (m *MachinePoolScope) GetShieldedInstanceConfigSpec() *compute.ShieldedInstanceConfig {
+	shieldedInstanceConfig := &compute.ShieldedInstanceConfig{
+		EnableSecureBoot:          false,
+		EnableVtpm:                true,
```

Comment:
Are these the defaults that GCP applies? Im afraid to make assumptions on bool values for upstream workers.

Reply:
For the InstanceSpec method for machine object also does similarly for reference https://github.com/newrelic-forks/cluster-api-provider-gcp/blob/main/cloud/scope/machine.go#L404-L410